src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -498,7 +575,38 @@ private Task CloseAsyncCore(WebSocketCloseStatus closeStatus, string? statusDesc
             }
         }
 
-        public override Task CloseOutputAsync(WebSocketCloseStatus closeStatus, string? statusDescription, CancellationToken cancellationToken) => throw new PlatformNotSupportedException();
+        public override Task CloseOutputAsync(WebSocketCloseStatus closeStatus, string? statusDescription, CancellationToken cancellationToken)
+        {
+            _writeBuffer = null;
+
+            WebSocketValidate.ValidateCloseStatus(closeStatus, statusDescription);
+
+            try
+            {
+                ThrowOnInvalidState(State, WebSocketState.Connecting, WebSocketState.Open, WebSocketState.CloseReceived, WebSocketState.CloseSent);
+            }
+            catch (Exception exc)
+            {
+                return Task.FromException(exc);
+            }
+            return CloseOutputAsyncCore(closeStatus, statusDescription, cancellationToken);
+        }
+
+        private Task CloseOutputAsyncCore(WebSocketCloseStatus closeStatus, string? statusDescription, CancellationToken cancellationToken)
+        {
+            try
+            {
+                _innerWebSocketCloseStatus = closeStatus;
+                _innerWebSocketCloseStatusDescription = statusDescription;
+                _innerWebSocket!.Invoke("close", (int)closeStatus, statusDescription);
```

Comment:
I'd just like to reiterate that not calling `onClose` manually here could result in a large delay. https://github.com/dotnet/runtime/issues/45468#issuecomment-741519725

Reply:
Brennan I saw that but is there anyway we can test this easily?

Here is the sequence of event on CloseOutputAsync that is proposed.

- Remove all event delegates not just on the close event or is close event enough?
- Call JavaScript `close()` method on `websocket`.
- Call JavaScript `onClose` event manually on the `websocket`.

I ask because if I leave the code as is the `onClose` will be called as well but the dotnet method returns immediately.

Reply:
We clear all events on the `websocket` (including `onClose`), call `websocket.close()` then call the user provided `onClose` method manually.

If you can write a test that blocks in `websocket.close()` and checks that the `onClose` method is still called then that would be good.

Reply:
Have to admit not sure how that would be done from the test harness for Websocket Client.  

Will add the code above.  Thanks for the feedback.

Reply:
Addressed if recent commits